### PR TITLE
[CIRCLE-11746] switch to flat orb.yml structure

### PR DESCRIPTION
**Pull request by johnswanson**

### CODE REVIEW

File: docs/commands.md [hunk 1]
```diff
@@ -69,22 +55,27 @@ CircleCI has several built-in commands available to all circleci.com customers a
...
-    type: string
-  overwrite:
-    default: false
-    type: boolean
-steps:
-  - run:
-      name: Deploy to S3
-      command: aws s3 sync << parameters.from >> << parameters.to >><<# parameters.overwrite >> --delete<</ parameters.overwrite >>"
+# orb.yml
+name: s3tools
```
Comment (ndintenfass) [inline docs/commands.md]: I don't think we'll have name in the orb definition

File: docs/executors.md [hunk 2]
```diff
@@ -110,30 +111,28 @@ jobs:
...
-- orbs
-  - foo-orb
-    - executors
-      - bar.yml
-  - baz-orb
-    - executors
-      - bar.yml
+```yaml
+# orb.yml
+name: foo-orb
```
Comment (ndintenfass) [inline docs/executors.md]: no names in orbs

File: docs/executors.md [hunk 3]
```diff
@@ -110,30 +111,28 @@ jobs:
...
+# config.yml
 jobs:
-  some_job:
+  some-job:
     executor: foo-orb/bar  # prefixed executor
-  some_other_job
-    executor:
-      name: baz-orb/bar    # prefixed executor
 ```
-Note that `foo-orb/bar` and `baz-orb/bar` are different executors. They both have the local name `bar`, but are independent executors living in different orbs.
```
Comment (ndintenfass) [inline docs/executors.md]: Why kill this?

File: docs/jobs.md [hunk 4]
```diff
@@ -49,25 +50,27 @@ workflows:
...
-    description: "To whom shall we say hello?"
-    default: "World"
-    type: string
-machine: true
-steps:
-  - say:
-      saywhat: "<< parameters.saywhat >>"
-```
+# .circleci/orbs/hello-orb/orb.yml
+name: hello-orb
```
Comment (ndintenfass) [inline docs/jobs.md]: no names in orbs

File: docs/jobs.md [hunk 5]
```diff
@@ -49,25 +50,27 @@ workflows:
...
+        default: "World"
+        type: string
+    machine: true
+    steps:
+      - say:
+          saywhat: "<< parameters.saywhat >>"
+commands:
   saywhat:
-    type: string
-steps:
```
Comment (ndintenfass) [inline docs/jobs.md]: This needs to stay, no?

File: docs/structure.md [hunk 6]
```diff
@@ -1,6 +1,8 @@
 # Orb Structure
 
-Each Orb is packaged and shared as a folder of files that follows a convenient convention for structuring configuration code. Each Orb also has its own namespace to use when invoking its elements. Orbs are authored with similar YAML syntax to the base CircleCI build configuration.
+Each Orb is packaged as a single file named `orb.yml`. Each Orb also has its
```
Comment (ndintenfass) [inline docs/structure.md]: This doesn't feel quite right -- we're saying that orbs are each a single yaml document, but the platform is agnostic to the file name. This is true above as well, but here it reads particularly oddly.

File: docs/structure.md [hunk 7]
```diff
@@ -9,22 +11,28 @@ Orbs are composed of one or more of the following elements, each of which repres
...
+An example orb:
 ```
-.circleci/orbs/foo/
-├── commands
-│   └── echo.yml
-├── jobs
-│   └── hello.yml
-├── orb.yml
+# .circleci/orbs/foo/orb.yml
+name: foo
```
Comment (ndintenfass) [inline docs/structure.md]: no names in orbs

File: docs/structure.md [hunk 8]
```diff
@@ -9,22 +11,28 @@ Orbs are composed of one or more of the following elements, each of which repres
...
+      - run: echo "hello world"
+jobs:
+  hello:
+    machine: true
+    steps:
+      - echo
 ```
 
-Our sample orb above has a single command name `echo` and a single job named `hello`. The `orb.yml` file has metadata about `foo` used mostly for auto-generating documentation such as a description, authorship info, etc.
+Here, an orb is placed in `.circleci/orbs/foo/orb.yml`. **NOTE:** _orbs inside
```
Comment (ndintenfass) [inline docs/structure.md]: Let's discuss this depending on whether we end up doing the local orbs thing.

File: docs/jobs.md (hunk 5 above)
Comment (johnswanson) [reply docs/jobs.md]: It's still there, just at a different level of indentation

**Pull request merged**